app/src/androidTest/java/ch/hikemate/app/ui/profile/ProfileScreenTest.kt

```diff
@@ -0,0 +1,150 @@
+      onSuccess(profile)
+    }
+    profileViewModel.getProfileById("1")
+
+    composeTestRule.onNodeWithTag(ProfileScreen.TEST_TAG_NAME).assertTextEquals(profile.name)
+    composeTestRule.onNodeWithTag(ProfileScreen.TEST_TAG_EMAIL).assertTextEquals(profile.email)
+    composeTestRule
+        .onNodeWithTag(ProfileScreen.TEST_TAG_HIKING_LEVEL)
+        .assertTextContains(
+            profile.hikingLevel.toString().lowercase(), substring = true, ignoreCase = true)
+    val current = LocalDate.now()
+    val currentDateDayOfMonth = current.dayOfMonth
+    val currentDateMonthText = current.month.toString().lowercase()
+    composeTestRule
+        .onNodeWithTag(ProfileScreen.TEST_TAG_JOIN_DATE)
+        .assertTextContains(currentDateDayOfMonth.toString(), substring = true, ignoreCase = true)
+    composeTestRule
+        .onNodeWithTag(ProfileScreen.TEST_TAG_JOIN_DATE)
+        .assertTextContains(currentDateMonthText, substring = true, ignoreCase = true)
+  }
+
+  @Test
+  fun checkExpertHikingLevel() {
+    val profile =
+        Profile(
+            id = "1",
+            name = "John Doe",
+            email = "john.doe@gmail.com",
+            hikingLevel = HikingLevel.EXPERT,
+            joinedDate = Timestamp.now())
+    `when`(profileRepository.getProfileById(any(), any(), any())).thenAnswer {
+      val onSuccess = it.getArgument<(Profile) -> Unit>(1)
+      onSuccess(profile)
+    }
+
+    profileViewModel.getProfileById("1")
+
+    composeTestRule
+        .onNodeWithTag(ProfileScreen.TEST_TAG_HIKING_LEVEL)
+        .assertTextContains(
+            profile.hikingLevel.toString().lowercase(), substring = true, ignoreCase = true)
+  }
+
+  @Test
+  fun checkBeginnerHikingLevel() {
+    val profile =
+        Profile(
+            id = "1",
+            name = "John Doe",
+            email = "john.doe@gmail.com",
+            hikingLevel = HikingLevel.BEGINNER,
+            joinedDate = Timestamp.now())
+    `when`(profileRepository.getProfileById(any(), any(), any())).thenAnswer {
+      @Suppress("UNCHECKED_CAST") val callback = it.arguments[1] as (Profile) -> Unit
+      callback(profile)
+    }
+    profileViewModel.getProfileById("1")
+
+    composeTestRule
+        .onNodeWithTag(ProfileScreen.TEST_TAG_HIKING_LEVEL)
+        .assertTextContains(
+            profile.hikingLevel.toString().lowercase(), substring = true, ignoreCase = true)
+  }
+
+  @Test
+  fun checkSidebarCanOpen() {
+    composeTestRule.onNodeWithTag(TEST_TAG_SIDEBAR_BUTTON).performClick()
+    composeTestRule.onNodeWithTag(TEST_TAG_DRAWER_CONTENT).assertIsDisplayed()
+  }
+
+  @Test
+  fun checkEditProfileButton() {
+    composeTestRule.onNodeWithTag(ProfileScreen.TEST_TAG_EDIT_PROFILE_BUTTON).performClick()
+    verify(navigationActions).navigateTo(Screen.EDIT_PROFILE)
+  }
+
+  @Test
+  fun checkSignOutButton() {
+    composeTestRule.onNodeWithTag(ProfileScreen.TEST_TAG_SIGN_OUT_BUTTON).performClick()
+    // TODO: Implement sign out logic
```

Comment:
Nitpick: This can still check if the viewmodel has been called for sign out, even if the repository is a dummy repository and doesn't really sign out after (checking the sign-out is correctly done is in the scope of the ViewModel and Model tests)

Reply:
Done!